kokoro/scripts/build/build_package.sh

```diff
@@ -43,41 +43,30 @@ export_to_sponge_config "PACKAGE_VERSION" "${PKG_VERSION}"
 ARCH="$(docker info --format '{{.Architecture}}')"
 ARTIFACT_REGISTRY="us-docker.pkg.dev"
 docker-credential-gcr configure-docker --registries="${ARTIFACT_REGISTRY}"
-CACHE_LOCATION="${ARTIFACT_REGISTRY}/stackdriver-test-143416/google-cloud-ops-agent-build-cache/ops-agent-cache:${DISTRO}_${ARCH}"
+CACHE="${ARTIFACT_REGISTRY}/stackdriver-test-143416/google-cloud-ops-agent-build-cache/ops-agent-cache"
 
-DOCKER_BUILDKIT=1 docker build . \
-  --cache-from="${CACHE_LOCATION}" \
+build_params=()
+if [[ -n "${KOKORO_GITHUB_PULL_REQUEST_NUMBER}" ]]; then  # Per-PR cache
+    build_params+=(--cache-from=type=registry,ref="${CACHE}:${DISTRO}_${ARCH}_PR${KOKORO_GITHUB_PULL_REQUEST_NUMBER}")
+    build_params+=(--cache-to=type=registry,ref="${CACHE}:${DISTRO}_${ARCH}_PR${KOKORO_GITHUB_PULL_REQUEST_NUMBER}",mode=max)
+fi
+if [[ "${KOKORO_ROOT_JOB_TYPE}" == "CONTINUOUS_INTEGRATION" ]]; then  # CI cache
+    build_params+=(--cache-to=type=registry,ref="${CACHE}:${DISTRO}_${ARCH}",mode=max)
```

Comment:
This will break our kokoro image prefetching. Which we could get rid of, but it has a cost.

As a historical note, your PR is effectively reversing https://github.com/GoogleCloudPlatform/ops-agent/pull/935, which did have some reasoning behind it...

Reply:
after offline discussion, we're going to try going forward with this.